src/lib/y2users/config.rb

```diff
@@ -92,6 +93,16 @@ def clone
       config
     end
 
+    # Merges elements (users and groups) with elements from another config
+    #
+    # @note The caller config is modified.
+    #
+    # @return [Config]
+    def merge(other, merger: nil)
+      merger ||= ConfigMergers::Replace
+      merger.new(self, other).merge
```

Comment:
NP: I still find better to pass arguments to merge instead of constructor, but not critical.

Reply:
IMHO, having the objects as internal attributes makes the implementation of the mergers more clear and easy to understand.